Source/WebKit/WebProcess/DigitalCredentials/DigitalCredentialsCoordinator.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "DigitalCredentialsCoordinator.h"
+
+#if ENABLE(WEB_AUTHN)
+#include "DigitalCredentialsCoordinatorProxyMessages.h"
+#include "FrameInfoData.h"
+#include "WebFrame.h"
+#include "WebPage.h"
+#include <WebCore/DigitalCredentialRequestOptions.h>
+#include <WebCore/LocalFrame.h>
+
+namespace WebKit {
+using namespace WebCore;
+
+DigitalCredentialsCoordinator::DigitalCredentialsCoordinator(WebPage& page)
+    : m_page(page)
+{
+}
+
+RefPtr<WebPage> DigitalCredentialsCoordinator::protectedPage() const
+{
+    return m_page.get();
+}
+
+void DigitalCredentialsCoordinator::requestDigitalCredential(const LocalFrame& frame, const DigitalCredentialRequestOptions& options, DigitalCredentialRequestCompletionHandler&& handler)
+{
+    RefPtr webFrame = WebFrame::fromCoreFrame(frame);
+    if (!webFrame)
+        return;
+
+    if (protectedPage())
+        protectedPage()->sendWithAsyncReply(Messages::DigitalCredentialsCoordinatorProxy::RequestDigitalCredential(webFrame->frameID(), webFrame->info(), options), WTFMove(handler));
```

Comment:
`page->`

We want to avoid repeated calls to `protectedPage()` as it causes ref-counting churn.